Library/Homebrew/diagnostic.rb

```diff
@@ -535,6 +535,24 @@ def check_user_path_3
         EOS
       end
 
+      def check_xdg_data_dirs
+        share = "#{HOMEBREW_PREFIX}/share"
+        homebrew_in_xdg_data_dirs = true
+        if ENV.key?("XDG_DATA_DIRS")
+          xdg_data_dirs = ENV["XDG_DATA_DIRS"].split(File::PATH_SEPARATOR)
+          xdg_data_dirs.length > 0 && homebrew_in_xdg_data_dirs = xdg_data_dirs.include?(share)
```

Comment:
I'm okay with that.

For my curiosity, which case didn't my code handle? Are you concerned about the case where `$XDG_DATA_DIRS` is set to the empty string?